compendium/ConfigurationAdmin/src/CMAsyncWorkService.cpp

```diff
@@ -83,20 +86,51 @@ namespace cppmicroservices
             {
                 if (threadpool)
                 {
-                    using Sig = void();
-                    using Result = boost::asio::async_result<decltype(task), Sig>;
-                    using Handler = typename Result::completion_handler_type;
-
-                    Handler handler(std::forward<decltype(task)>(task));
+                    boost::asio::post(threadpool->get_executor(), std::move(task));
+                }
+            }
 
-                    boost::asio::post(threadpool->get_executor(),
-                                      [handler = std::move(handler)]() mutable { handler(); });
+            // createStrand returns a new AsyncWorkService instance posting to a strand
+            std::shared_ptr<AsyncWorkService>
+            createStrand() override
+            {
+                if (!threadpool)
+                {
+                    return nullptr;
                 }
+                auto strand = std::make_shared<Strand>(threadpool->get_executor());
+                // Pass shared_ptr to threadpool to keep it alive as long as strand is alive
+                return std::make_shared<StrandAsyncWorkService>(strand, logger);
             }
 
           private:
             std::shared_ptr<boost::asio::thread_pool> threadpool;
             std::shared_ptr<cppmicroservices::logservice::LogService> logger;
+
+            // Inner class for strand-based async work service
+            class StrandAsyncWorkService : public AsyncWorkService
+            {
+              public:
+                StrandAsyncWorkService(std::shared_ptr<Strand> strand_,
+                                       std::shared_ptr<cppmicroservices::logservice::LogService> logger_)
+                    : strand(std::move(strand_))
+                    , logger(std::move(logger_))
+                {
+                }
+
+                void
+                post(std::packaged_task<void()>&& task) override
+                {
+                    if (strand)
+                    {
+                        boost::asio::post(*strand, std::move(task));
```

Comment:
Ditto here - wrap the `task` in a lambda to stop `asio` from trying to obtain the future?